templates/packer/ansible/roles/common/tasks/rpmbuild.yml

```diff
@@ -0,0 +1,95 @@
+---
+
+- name: Create rpmbuild directory
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+    - /tmp/rpmbuild
+    - /tmp/rpmbuild/SOURCES
+    - /tmp/rpmbuild/RPMS
+
+- name: Download osbuild-composer archive
+  get_url:
+    url: "https://github.com/gundersanne/osbuild-composer/archive/{{ COMPOSER_COMMIT }}.tar.gz"
+    dest: "/tmp/rpmbuild/SOURCES/osbuild-composer-{{ COMPOSER_COMMIT }}.tar.gz"
+
+- name: Download osbuild-composer spec
+  get_url:
+    url: "https://raw.githubusercontent.com/gundersanne/osbuild-composer/{{ COMPOSER_COMMIT }}/osbuild-composer.spec"
+    dest: /tmp/osbuild-composer.spec
+
+- name: Download osbuild archive
+  get_url:
+    url: "https://github.com/gundersanne/osbuild/archive/{{ OSBUILD_COMMIT }}.tar.gz"
+    dest: "/tmp/rpmbuild/SOURCES/osbuild-{{ OSBUILD_COMMIT }}.tar.gz"
+
+- name: Download osbuild spec
+  get_url:
+    url: "https://raw.githubusercontent.com/gundersanne/osbuild/{{ OSBUILD_COMMIT }}/osbuild.spec"
+    dest: /tmp/osbuild.spec
+
+- name: Install build tools
+  package:
+    name:
+      - rpm-build
+      - mock
+    state: present
+
+- name: Make osbuild srpm
+  command: >-
+    rpmbuild -bs
+    --define "commit {{ OSBUILD_COMMIT }}"
+    --define "_topdir /tmp/rpmbuild"
+    --define "_srpmfilename %%{NAME}.srpm"
+    /tmp/osbuild.spec
+
+- name: Mockbuild osbuild
+  shell: >-
+      mock
+      --rebuild
+      --define "commit {{ OSBUILD_COMMIT }}"
+      --define "_rpmfilename %%{NAME}.rpm"
+      --resultdir /tmp/rpmbuild/RPMS
+      /tmp/rpmbuild/SRPMS/osbuild-*.src.rpm
+
+- name: Make osbuild-composer srpm
+  command: >-
+    rpmbuild -bs
+    --define "commit {{ COMPOSER_COMMIT }}"
+    --define "_topdir /tmp/rpmbuild"
+    --define "_srpmfilename %%{NAME}.srpm"
+    /tmp/osbuild-composer.spec
+
+
+- name: Mockbuild osbuild-composer
+  shell: >-
+      mock
+      --rebuild
+      --define "commit {{ COMPOSER_COMMIT }}"
+      --define "_rpmfilename %%{NAME}.rpm"
+      --resultdir /tmp/rpmbuild/RPMS
+      /tmp/rpmbuild/SRPMS/osbuild-composer-*.src.rpm
+
+- name: Install artifacts
+  command: >-
+    dnf install -y
+    /tmp/rpmbuild/RPMS/osbuild.rpm
```

Comment:
I have two ideas:

1) don't install `osbuild-tools` here.
2) put everything in a local repository using `createrepo_c`, add a repofile to `/etc/yum.repos.d` (with an appropriate priority) and install just `osbuild-composer-worker`, everything else will be pulled using deps.